client/src/elm/Measurement/Model.elm

```diff
@@ -1868,3 +1868,32 @@ type alias NutritionCaringForm =
 emptyNutritionCaringForm : NutritionCaringForm
 emptyNutritionCaringForm =
     NutritionCaringForm Nothing Nothing Nothing
+
+
+type alias OngoingTreatmentReviewForm =
+    { takenAsPrescribed : Maybe Bool
+    , missedDoses : Maybe Bool
+    , feelingBetter : Maybe Bool
+    , sideEffects : Maybe Bool
+    , reasonForNotTaking : Maybe ReasonForNotTaking
+    , reasonForNotTakingDirty : Bool
+    , totalMissedDoses : Maybe Int
+    , totalMissedDosesDirty : Bool
+    , adverseEvents : Maybe (List AdverseEvent)
+    , adverseEventsDirty : Bool
+    }
+
+
+emptyOngoingTreatmentReviewForm : OngoingTreatmentReviewForm
+emptyOngoingTreatmentReviewForm =
+    { takenAsPrescribed = Nothing
+    , missedDoses = Nothing
+    , feelingBetter = Nothing
+    , sideEffects = Nothing
+    , reasonForNotTaking = Nothing
+    , reasonForNotTakingDirty = False
+    , totalMissedDoses = Nothing
+    , totalMissedDosesDirty = False
+    , adverseEvents = Nothing
+    , adverseEventsDirty = False
```

Comment:
How does this `Dirty` concept work? As I see, empty data is not dirty, upon input, it becomes dirty? But does this become non-dirty afterwards at some point?

Reply:
@AronNovak 

We have following situation:
Form is the UI, and on refresh all it's fields loose set values.
When we store the form, Value is created - representation of stored data for back-end.
To allow editing saved form, we need to combine form with Value, and present the data.
For correct presentation we have `form <=> value` functions.

Dirty fields are used when we have a conditional input, which is edited and need to get empty value.
For example:
1. Question: 'Did you administer medication?'
2. If 1 answer is No, the we ask for the reason and list several options.

So, we have a form with field for q1 and q2.
On first run, user answers q1 No, and fills the reason for Q2.
On save, value is created with 2 fields - v1 and v2.
Now page is refreshed, and user decides to edit the answer to q1 - enters Yes.
If we do nothing, on save `form => value` function will result in same v2 we had previously, while it needs to be empty.
To overcome this, we use dirty field for q2. So, when q1 is set to Yes, we set q2 to empty value and q2Dirty set to true, which will indicate for `form => value` function to force use of q2 from form.